sky/sdk/lib/widgets/snack_bar.dart

```diff
@@ -29,18 +35,58 @@ class SnackBarAction extends Component {
   }
 }
 
-class SnackBar extends Component {
+class SnackBar extends AnimatedComponent {
 
   SnackBar({
     String key,
     this.content,
-    this.actions
+    this.actions,
+    this.showing,
+    this.onHidden
```

Comment:
In Drawer and PopupMenu, we introduced an enum to differentiate this state more clearly from `showing`.